### PR TITLE
[Documentation][Refund] Add cookbook about refund customization with improvements

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | 1.9
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | bases on https://github.com/Sylius/Sylius/pull/12717 and waits for merge https://github.com/Sylius/RefundPlugin/pull/320
| License         | MIT

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]

File: docs/cookbook/payments/custom-field-on-refund-payment.rst [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Event;
+
+    use Sylius\RefundPlugin\Event\UnitsRefunded as BaseUnitsRefunded;
+
+    class UnitsRefunded extends BaseUnitsRefunded
```
Comment (Zales0123) [inline docs/cookbook/payments/custom-field-on-refund-payment.rst:362]: Could be final, I suppose 🖖

File: docs/cookbook/payments/custom-field-on-refund-payment.rst [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+
+            $refundedTotal = 0;
+            $refundedTotal += $this->orderUnitsRefunder->refundFromOrder($command->units(), $orderNumber);
+            $refundedTotal += $this->orderShipmentsRefunder->refundFromOrder($command->shipments(), $orderNumber);
+
+            /** @var string|null $currencyCode */
+            $currencyCode = $order->getCurrencyCode();
+            Assert::notNull($currencyCode);
+
+            // Dispatching a new event
```
Comment (Zales0123) [inline docs/cookbook/payments/custom-field-on-refund-payment.rst:316]: Don't say 😄

File: docs/cookbook/payments/custom-field-on-refund-payment.rst [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+    use Sylius\RefundPlugin\Event\RefundPaymentGenerated;
+    use Sylius\RefundPlugin\Event\UnitsRefunded;
+    use Sylius\RefundPlugin\Factory\RefundPaymentFactoryInterface;
+    use Sylius\RefundPlugin\ProcessManager\UnitsRefundedProcessStepInterface;
+    use Sylius\RefundPlugin\Provider\RelatedPaymentIdProviderInterface;
+    use Sylius\RefundPlugin\StateResolver\OrderFullyRefundedStateResolverInterface;
+    use Symfony\Component\Messenger\MessageBusInterface;
+    use Webmozart\Assert\Assert;
+
+    final class RefundPaymentProcessManager implements UnitsRefundedProcessStepInterface
```
Comment (Zales0123) [inline docs/cookbook/payments/custom-field-on-refund-payment.rst:414]: It's sad, we cannot decorate it or override some smaller services :( Perhaps we should thing about separating some concepts from the original handler to other services

File: docs/cookbook/payments/custom-field-on-refund-payment.rst [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+
+**7. Display the new field on the refund payment:**
+
+And as the last step, we need to overwrite the template ``_refundPayments.html.twig`` from Refund Plugin.
+Copy the entire ``_refundPayments.html.twig`` to ``templates/bundles/SyliusRefundPlugin/Order/Admin/_refundPayments.html.twig``:
+
+.. code-block:: bash
+
+    mkdir -p templates/bundles/SyliusRefundPlugin/Order/Admin
+    cp vendor/sylius/refund-plugin/src/Resources/views/Order/Admin/_refundPayments.html.twig templates/bundles/SyliusRefundPlugin/Order/Admin/
```
Comment (Zales0123) [inline docs/cookbook/payments/custom-field-on-refund-payment.rst:516]: Again, it's a clue for us, that maybe we could throw some UI event there 🚀

**Review by Zales0123 — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Grzegorz! :1st_place_medal:

**Pull request merged**